src/cpu/simple/base.cc

```diff
@@ -383,8 +386,10 @@ BaseSimpleCPU::preExecute()
     //If we decoded an instruction this "tick", record information about it.
     if (curStaticInst) {
 #if TRACING_ON
+      if (debug::ExecEnable) { //getInstRecord does nothing unless ExecEnable
```

Comment:
This looks wrong to me. The getInstRecord is called irrespective of the tracer (the method is part of the interface). The Exec tracer (enebled by ExecEnable) is only one of them...

Reply:
The `getInstRecord` function body is:

```cpp
if (!debug::ExecEnable)
        return NULL;

    return new InstPBTraceRecord(*this, when, tc, si, pc, mi);
```

My unease at this change is more that it couples the logic of this and the logic of `getInstRecord` if we were to update `getInstRecord` this could break without us knowing.

Essentially, the optimization here is just saving a function call which this tool seems to think is worth doing. 

I think i'd probably vote to remove this change too but mostly due to it being a bit brittle.

Reply:
The getInstRecord is a virtual method implemented by every tracer we support in gem5.

See for example:

https://github.com/gem5/gem5/blob/stable/src/arch/arm/tracers/tarmac_tracer.cc#L100

The ExecEnable is just specific to ONE of them (the ExecTracer).
With this change basically we would break all existing tracers not using the ExecEnable flag.